src/main/java/org/folio/circulation/resources/RequestFromRepresentationService.java

```diff
@@ -275,11 +276,53 @@ private CompletableFuture<Result<Request>> fetchItemAndLoanForRecallTlrRequest(R
       .thenComposeAsync(r -> r.after(when(loan -> ofAsync(() -> loan == null),
         ignored -> ofAsync(requestQueue::getTheLeastRecalledLoan), result -> ofAsync(() ->  result))))
       .thenApply(resultLoan -> resultLoan.map(request::withLoan))
-      .thenComposeAsync(requestResult -> requestResult.combineAfter(
-        r -> itemRepository.fetchFor(r.getLoan()), Request::withItem))
+      .thenApply(this::refuseWhenNoAllowedItemAndLoan)
+      .thenApply(this::fetchAllowedItemIfNoLoanExists)
+      .thenCompose(r -> r.after(this::fetchItemForLoan))
       .thenComposeAsync(requestResult -> requestResult.combineAfter(
         this::getUserForExistingLoan, this::addUserToLoan))
-      .thenApply(r -> errorHandler.handleValidationResult(r, INSTANCE_DOES_NOT_EXIST, request));
+      .thenApply(res -> errorHandler.handleValidationResult(res, INSTANCE_DOES_NOT_EXIST, request));
+  }
+
+  private CompletableFuture<Result<Request>> fetchItemForLoan(Request request) {
+    if (request.getLoan() == null) {
+      return ofAsync(() -> request);
+    }
+    return itemRepository.fetchFor(request.getLoan())
+      .thenApply(r -> r.map(request::withItem));
+  }
+
+  private Result<Request> fetchAllowedItemIfNoLoanExists(Result<Request> request) {
+    return request.next(req -> {
+      Item item = shouldHaveItemsWithAllowedStatusesForRecall(req);
+      if (req.getLoan() == null && item != null) {
+        return Result.succeeded(req.withItem(item));
+      }
+      return of(() -> req);
+    });
+  }
+
+  private Result<Request> refuseWhenNoAllowedItemAndLoan(Result<Request> request) {
+    return request.next(this::validateIfAllowedItemAndLoanExist)
+      .mapFailure(err -> errorHandler.handleValidationError(err,
+        ATTEMPT_TO_RECALL_WITHOUT_LOAN_AND_ALLOWED_ITEMS, request));
+  }
+
+  private Result<Request> validateIfAllowedItemAndLoanExist(Request request) {
+    Item item = shouldHaveItemsWithAllowedStatusesForRecall(request);
+    if (request.getLoan() == null && item == null) {
+      return failedValidation("Request doesn't have loan and items with allowed statuses",
+        "loan", null);
+    }
+    return of(() -> request);
+  }
+
+  private Item shouldHaveItemsWithAllowedStatusesForRecall(Request request) {
+    return request.getInstanceItems().stream()
+      .filter(item -> List.of("Awaiting delivery", "Paged", "Awaiting pickup")
+        .contains(item.getStatus().getValue()))
```

Comment:
Extract the list into a variable please - the object is being created for every stream element